daemons/base.py

```diff
@@ -231,67 +265,64 @@ def parse_params(self, params):
             args = ()
         return args, kwargs
 
-    async def handle_request(self, request):
-        auth = request.headers.get("Authorization")
-        user, password = self.decode_auth(auth)
-        if not (user == self.LOGIN and password == self.PASSWORD):
-            return web.json_response({"jsonrpc": "2.0", "error": {"code": -32600, "message": "Unauthorized"}, "id": None})
-        if not LEGACY_AIOHTTP:
-            data = await request.json(content_type=None)  # aiohttp 4.0
-        else:
-            data = await request.json()
-        method = data.get("method")
-        id = data.get("id", None)
-        params = data.get("params", [])
+    async def get_handle_request_params(self, request):
+        data = await (request.json() if LEGACY_AIOHTTP else request.json(content_type=None))
+        method, id, params = data.get("method"), data.get("id", None), data.get("params", [])
+        error = None if method else JsonResponse(code=-32601, error="Procedure not found", id=id)
         args, kwargs = self.parse_params(params)
-        xpub = kwargs.pop("xpub", None)
-        if not method:
-            return web.json_response(
-                {"jsonrpc": "2.0", "error": {"code": -32601, "message": "Procedure not found."}, "id": id}
-            )
+        return id, method, args, kwargs, error
+
+    async def get_exec_method(self, cmd, id, req_method):
+        exec_method, custom = (
+            (self.supported_methods[req_method], True)
+            if req_method in self.supported_methods
+            else (getattr(cmd, req_method, JsonResponse(code=-32601, error="Procedure not found", id=id)), False)
+        )
+        error = exec_method if isinstance(exec_method, JsonResponse) else None
+        return exec_method, custom, error
+
+    async def get_exec_result(self, req_method, req_args, req_kwargs, exec_method, custom, **kwargs):
+        xpub = req_kwargs.pop("xpub", None)
+        if custom:
+            exec_method = functools.partial(exec_method, wallet=xpub)
+        else:
+            if self.NEW_ELECTRUM and self.electrum.commands.known_commands[req_method].requires_wallet:
+                wallet, config = kwargs.get("wallet"), kwargs.get("config")
```

Comment:
Why not just passing wallet and config as params?

Reply:
Maybe "wallet" and "config" are too specific for this method. If a derived class overrides this method, it may not need these two arguments

Reply:
Hmm, for some coins implementations not based on electrum. I agree